libavpipe/src/avpipe_xc.c

```diff
@@ -350,6 +338,60 @@ check_stream_index(
     return eav_success;
 }
 
+/*
+ * Validate input stream against XC parameters.
+ * Called after avformat_find_stream_info
+ */
+static int
+check_input_stream(
+    xcparams_t *params,
+    coderctx_t *decoder_context) {
+
+    int rc;
+
+    if (!decoder_context->format_context->iformat &&
+        !decoder_context->format_context->iformat->name) {
+        elv_err("Failed to open input stream properly - no format name");
+        return eav_open_input;
+    }
+
+    switch(decoder_context->live_proto) {
+        case avp_proto_mpegts:
+        case avp_proto_srt:
+            if (strcmp(decoder_context->format_context->iformat->name, "mpegts")) {
+                elv_err("Unsupported live source proto=%d container=%s",
+                    decoder_context->live_proto, decoder_context->format_context->iformat->name);
+                return eav_open_codec;
+            }
+            break;
+        case avp_proto_rtmp:
+            if (strcmp(decoder_context->format_context->iformat->name, "flv")) {
+                elv_err("Unsupported live source proto=%d container=%s",
+                    decoder_context->live_proto, decoder_context->format_context->iformat->name);
+                return eav_open_codec;
+            }
+            break;
+        case avp_proto_rtp:
+            if (decoder_context->format_context->priv_data) {
+                int64_t payload_type;
```

Comment:
nit: Just name the constant instead of adding a comment after checking

```suggestion
                int64_t payload_type;
                int64_t rtp_payload_type_mpegts = 33;
```